boards/nxp/frdm_mcxn947/board.c

```diff
@@ -299,8 +299,21 @@ static int frdm_mcxn947_init(void)
 #endif
 
 #if DT_NODE_HAS_STATUS(DT_NODELABEL(lptmr0), okay)
+
+
+#if DT_PROP(DT_NODELABEL(lptmr0), clk_source) == 0x0
+	CLOCK_SetupClockCtrl(kCLOCK_FRO12MHZ_ENA);
+#elif DT_PROP(DT_NODELABEL(lptmr0), clk_source) == 0x1
 	CLOCK_SetupClk16KClocking(kCLOCK_Clk16KToVsys);
-#endif
+#elif DT_PROP(DT_NODELABEL(lptmr0), clk_source) == 0x2
+	CLOCK_SetupOsc32KClocking(kCLOCK_Osc32kToVsys);
+#elif DT_PROP(DT_NODELABEL(lptmr0), clk_source) == 0x3
```

Comment:
somewhat magic numbers here, and in DT, is there something we can do about that

Reply:
I agree. Is there any defines in the SDK? Or if there isn't lets put a comment block highlighting what the different values are (for future you and others).

Reply:
issue with the comment block is that the numbers will also be magic in DT, in a way, the numbers are kind of explained here because the code is pretty clearly showing what the clock sources are, but user shouldnt have to find this file and these lines of code comment or not to know what number to put in DT

Reply:
Updated the binding file and added a comment block here, please let me know if this helps.